packages/thirdweb/src/contract/deployment/utils/create-2-factory.ts

```diff
@@ -88,20 +104,21 @@ export async function deployCreate2Factory(options: ClientAndChainAndAccount) {
   const { client, chain, account } = options;
   const enforceEip155 = await isEIP155Enforced(options);
   const chainId = options.chain.id;
-  const custom = CUSTOM_GAS_FOR_CHAIN[chainId.toString()];
   const eipChain = enforceEip155 ? chainId : 0;
 
-  const deploymentInfo = custom
-    ? await _getCreate2FactoryDeploymentInfo(eipChain, {
-        gasPrice: custom.gasPrice,
-        gasLimit: custom.gasLimit,
-      })
-    : await _getCreate2FactoryDeploymentInfo(eipChain, {});
   const rpcRequest = getRpcClient({
     client: client,
     chain,
   });
 
+  const gasPriceFetched = await getGasPrice(options);
+  const bin =
+    CUSTOM_GAS_BINS.find((e) => e >= gasPriceFetched) || gasPriceFetched;
```

Comment:
this relies on the order of the gas bins array which is a little scary but probably ok?